dashboard/app/controllers/projects_controller.rb

```diff
@@ -1,14 +1,24 @@
 class ProjectsController < ApplicationController
-  before_filter :authenticate_user!
+  before_filter :authenticate_user!, except: [:show, :edit, :readonly]
+  before_action :set_level, only: [:show, :edit, :readonly]
```

Comment:
this was created using levels_controller.rb as inspriration. I'm not sure if it needs some of that other boiler plate as well? Maybe check_authorization?